Security: don't allow for user-specified command execution for
compression

--- a/tcpdump.c
+++ b/tcpdump.c
@@ -230,7 +230,6 @@
 #endif
 static int Wflag;			/* recycle output files after this number of files */
 static int WflagChars;
-static char *zflag = NULL;		/* compress each savefile using a specified command (like gzip or bzip2) */
 static int timeout = 1000;		/* default timeout = 1000 ms = 1 s */
 #ifdef HAVE_PCAP_SET_IMMEDIATE_MODE
 static int immediate_mode;
@@ -1876,10 +1875,6 @@
 			}
 			break;
 #endif
-		case 'z':
-			zflag = optarg;
-			break;
-
 		case 'Z':
 			username = optarg;
 			break;
@@ -2537,7 +2532,7 @@
 	}
 
 #ifdef HAVE_CAPSICUM
-	cansandbox = (VFileName == NULL && zflag == NULL);
+	cansandbox = (VFileName == NULL);
 #ifdef HAVE_CASPER
 	cansandbox = (cansandbox && (ndo->ndo_nflag || capdns != NULL));
 #else
@@ -2805,60 +2800,7 @@
 	infoprint = 0;
 }
 
-#if defined(HAVE_FORK) || defined(HAVE_VFORK)
-#ifdef HAVE_FORK
-#define fork_subprocess() fork()
-#else
-#define fork_subprocess() vfork()
-#endif
 static void
-compress_savefile(const char *filename)
-{
-	pid_t child;
-
-	child = fork_subprocess();
-	if (child == -1) {
-		fprintf(stderr,
-			"compress_savefile: fork failed: %s\n",
-			pcap_strerror(errno));
-		return;
-	}
-	if (child != 0) {
-		/* Parent process. */
-		return;
-	}
-
-	/*
-	 * Child process.
-	 * Set to lowest priority so that this doesn't disturb the capture.
-	 */
-#ifdef NZERO
-	setpriority(PRIO_PROCESS, 0, NZERO - 1);
-#else
-	setpriority(PRIO_PROCESS, 0, 19);
-#endif
-	if (execlp(zflag, zflag, filename, (char *)NULL) == -1)
-		fprintf(stderr,
-			"compress_savefile: execlp(%s, %s) failed: %s\n",
-			zflag,
-			filename,
-			pcap_strerror(errno));
-#ifdef HAVE_FORK
-	exit(S_ERR_HOST_PROGRAM);
-#else
-	_exit(S_ERR_HOST_PROGRAM);
-#endif
-}
-#else  /* HAVE_FORK && HAVE_VFORK */
-static void
-compress_savefile(const char *filename)
-{
-	fprintf(stderr,
-		"compress_savefile failed. Functionality not implemented under your system\n");
-}
-#endif /* HAVE_FORK && HAVE_VFORK */
-
-static void
 dump_packet_and_trunc(u_char *user, const struct pcap_pkthdr *h, const u_char *sp)
 {
 	struct dump_info *dump_info;
@@ -2903,12 +2845,6 @@
 			 * Close the current file and open a new one.
 			 */
 			pcap_dump_close(dump_info->pdd);
-
-			/*
-			 * Compress the file we just closed, if the user asked for it
-			 */
-			if (zflag != NULL)
-				compress_savefile(dump_info->CurrentFileName);
 
 			/*
 			 * Check to see if we've exceeded the Wflag (when
@@ -3009,13 +2945,6 @@
 			 */
 			pcap_dump_close(dump_info->pdd);
 
-			/*
-			 * Compress the file we just closed, if the user
-			 * asked for it.
-			 */
-			if (zflag != NULL)
-				compress_savefile(dump_info->CurrentFileName);
-
 			Cflag_count++;
 			if (Wflag > 0) {
 				if (Cflag_count >= Wflag)
@@ -3230,5 +3159,5 @@
 "\t\t[ --time-stamp-precision precision ] [ --micro ] [ --nano ]\n");
 #endif
 	(void)fprintf(f,
-"\t\t[ -z postrotate-command ] [ -Z user ] [ expression ]\n");
+"\t\t[ -Z user ] [ expression ]\n");
 }
